### ISSUE TITLE
`prefer-default-export` fails on Typescript type declaration

### ISSUE THREAD

**Issue by sharmilajesupaul** [benmosher/eslint-plugin-import#1332]
`import/prefer-default-export` erroneously warns on files that contain a single exported type.

eg: 
```
// foo.ts 

export type Foo = { // `prefer-default-export` fails on this line
 bar: string;
};
```

Related issue: https://github.com/benmosher/eslint-plugin-import/issues/1164

**Comment by thisissami** [comment]
Not sure if this helps or not, but this was addressed once a while back: https://github.com/benmosher/eslint-plugin-import/pull/639 seems like this bug was reintroduced.

**Comment by ljharb** [comment]
That was for flow, this is for typescript.

**Comment by thisissami** [comment]
@ljharb my assumption based off of https://github.com/benmosher/eslint-plugin-import/issues/1164 is that it is broken for both `typesript` and `flow`

**Comment by thisissami** [comment]
and both `typescript` and `flow` use the identical `export type insert_type_name_here` syntax, so unless the parser was explicitly looking for `// @flow` at the top of a file - it should work identically for each.

**Comment by mkaraula** [comment]
👋 are there any news on this?
